metadata-ws/src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/events/TaxonomyEventHandler.java

```diff
@@ -77,6 +77,9 @@ public Taxonomy importTaxonomyTree(Taxonomy taxonomy) throws Exception {
                 long taxIdLong = Long.parseLong(taxId);
                 Taxonomy taxonomyParent = importTaxonomyTree(new Taxonomy(taxIdLong));
                 taxonomy.setTaxonomyForRank(taxonomyParent, rankEnum.toString());
+            } else if (taxonomy.getRank().equals(RANK.SPECIES.name) && taxonomy.getTaxonomySpecies() == null) {
+                // self-reference at species level
+                taxonomy.setTaxonomySpecies(taxonomy);
             }
         }
```

Comment:
Shouldn't this method include `return taxonomy` at the end? Or am I missing something?

Reply:
I think "return taxonomyRepository.save(taxonomy);" is at the end

Reply:
You're right! I just missed it somehow